HWs/Test3/task1/main.c

```diff
@@ -0,0 +1,126 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include <string.h>
+#include <stdlib.h>
+
+#define binarySize 256
+
+int roundUp(int firstNumber, int secondNumber)
+{
+    int value = firstNumber / secondNumber;
+    return value * secondNumber == firstNumber ? value : value + 1;
+}
+
+bool isCorrectBinaryNumber(char *number)
+{
+    int stringLength = strlen(number);
+    for (int i = 0; i < stringLength; ++i)
+    {
+        if (number[i] != '0' && number[i] != '1')
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+char triadTransformation(int number[])
+{
+    int power = 1;
+    int result = 0;
+    for (int i = 0; i < 3; ++i)
+    {
+        result += power * number[i];
+        power <<= 1;
+    }
+
+    return result + '0';
+}
+
+char *binaryToOctal(char *binaryNumber)
+{
+    int binaryNumberLength = strlen(binaryNumber) - 1;
+    int lengthOctalNumber = roundUp(binaryNumberLength + 1, 3);
+    char *octalNumber = calloc(lengthOctalNumber + 1, sizeof(char));
+    if (octalNumber == NULL)
+    {
+        return NULL;
+    }
+    octalNumber[lengthOctalNumber--] = '\0';
+    int counter = 0;
+    int triad[3] = {0};
+    for (int i = binaryNumberLength; i >= 0; --i)
+    {
+        if (binaryNumber[i] == '1')
+        {
+            triad[counter] = 1;
+        }
+        else
+        {
+            triad[counter] = 0;
+        }
```

Comment:
```suggestion
       triad[counter++] = binaryNumber[i] - '0';
```
Зачем так многословно? Ведь символы - это числа
(-0.2)